frontend-ssr/components/cli_git/GitCli_CC.tsx

```diff
@@ -0,0 +1,145 @@
+  ERROR = 'error',
+  SUCCESS = 'success'
+}
+
+class matchCommandMessageClassComponent extends React.Component<Props, State, Message> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      currentIndex: 0,
+      messages: [],
+      btnMessage: 'Next',
+      loading: false,
+    };
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+
+    const formEl = event.target;
+    const gitCommand = event.target.gitCommand.value;
+    const { commands, successMessages } = this.props;
+    const { currentIndex } = this.state;
+
+    let isMatch = false;
+
+    if (commands[currentIndex] instanceof RegExp) {
+      isMatch = (commands[currentIndex] as RegExp).test(gitCommand);
+    } else {
+      isMatch = commands[currentIndex] === gitCommand.trim().toLowerCase();
+    }
+
+    const commandMessage: Message = {
+      text: gitCommand,
+      type: MessageType.COMMAND,
+      time: Date.now(),
+      id: uuid(),
+    };
+
+    let matchCommandMessage: Message;
+
+    if (!isMatch) {
+      matchCommandMessage = {
+        text: 'This is not a valid command',
+        type: MessageType.ERROR,
+        time: Date.now(),
+        id: uuid(),
+      };
+    } else {
+      matchCommandMessage = {
+        text: successMessages[currentIndex],
+        type: MessageType.SUCCESS,
+        time: Date.now(),
+        id: uuid(),
+      };
+    }
+
+    if (!isMatch) {
+      this.setState(({ messages }) => ({
+        messages: [...messages, commandMessage, matchCommandMessage],
+        btnMessage: 'Next',
+      }));
+    } else if (currentIndex + 1 === commands.length) {
+      this.setState(({ messages }) => ({
+        messages: [...messages, commandMessage, matchCommandMessage],
+        currentIndex: 0,
+        btnMessage: 'Done',
+        loading: true,
+      }));
+    } else {
+      this.setState(({ messages }) => ({
+        messages: [...messages, commandMessage, matchCommandMessage],
+        currentIndex: currentIndex + 1,
+      }));
+    }
+    formEl.reset();
+  };
+
+  render() {
+    const { messages, btnMessage, loading } = this.state;
+
+    return (
+      <>
+        <div className={styles.commands}>
+          <div className={styles.commands__cli}>
+            {messages.map((message) => (
+              <p key={message.id} className={styles[message.type]}>
+                <span>{message.text}</span>
+                <span>
+                  {new Date(message.time).toLocaleString()}
+                </span>
+              </p>
+            ))}
+          </div>
+          <form
+            className={styles.form}
+            onSubmit={this.handleSubmit}
+          >
+            <input
+              type="text"
+              name="gitCommand"
+              spellCheck="false"
+              required
+            />
+            <span className={styles.blinking__cursor} />
+            <button
+              className="btn btn--blue"
+              type="submit"
+              disabled={loading}
```

Comment:
As mentioned before, we don't need another variable for this. We can simply check if `currentIndex === commands.length`. This means we've reached the end. 

Also, forgot to tell you in the last review. Add a `readOnly` attribute to the input when we've finished everything, so that you cannot write there.